### PR TITLE
Backport (4.3) for installing cluster on Azure UPI 

### PR DESCRIPTION
**Pull request by codyhoag**
This backports the 4.4 documentation for installing a cluster on Azure with UPI (reviewed and merged in #19831) to 4.3. This was originally completed for [CORS-1285](https://issues.redhat.com/browse/CORS-1285) (specifically - [OSDOCS-775](https://issues.redhat.com/browse/OSDOCS-775)). This PR is in preparation for the 4.3 code backport being tracked in [CORS-1407](https://issues.redhat.com/browse/CORS-1407).

### CODE REVIEW

**Comment by codyhoag** [comment]
Preview: http://file.rdu.redhat.com/~choag/040220/azure-upi-4.3/installing/installing_azure/installing-azure-user-infra.html

**Comment by codyhoag** [comment]
Due to our docs support policy, this will require several ACKs from stakeholders before merging.

@katherinedube @sferich888 @reestr @xltian @vikram-redhat can you ACK this documentation backport?

@fabianofranz @e-tienne can you reconfirm this? This is a direct cherrypick from the 4.4 docs, since no code or procedures changed.

Thanks all!

**Comment by katherinedube** [comment]
ACK - this backport is in support of a customer RFE.

**Comment by fabianofranz** [comment]
LGTM, thanks @codyhoag!

**Comment by xltian** [comment]
@e-tienne  will review and ack from QE, before ack, QE needs to make sure this has been tested in 4.3 and added in 4.3 CI firstly, I don't think this is tested so far.

**Comment by e-tienne** [comment]
LGTM as well

Confirmed working latest nightly. Do we also have a scenario ready for PROW CI?

File: modules/installation-azure-user-infra-uploading-rhcos.adoc [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
++
+----
+$ export ACCOUNT_KEY=`az storage account keys list -g ${RESOURCE_GROUP} --account-name ${CLUSTER_NAME}sa --query "[0].value" -o tsv`
+----
+
+. Choose the {op-system} version to use and export the URL of its VHD to an
+environment variable:
++
+----
+$ export VHD_URL=`curl -s https://raw.githubusercontent.com/openshift/installer/release-4.4/data/data/rhcos.json | jq -r .azure.url`
```
Comment (e-tienne) [inline modules/installation-azure-user-infra-uploading-rhcos.adoc:47]: Hello @codyhoag , I tested last week, but I noticed today that I did not use the same RHCOS image that you have in this PR. My understanding is that you would need to use a 4.3 RHOS image with 4.3 so I used `export VHD_URL=`curl -s https://raw.githubusercontent.com/openshift/installer/release-4.3/data/data/rhcos.json | jq -r .azure.url`

Comment (codyhoag) [reply modules/installation-azure-user-infra-uploading-rhcos.adoc]: @e-tienne good catch! I just updated the RHCOS image to 4.3

**Comment by codyhoag** [comment]
This has received the appropriate Acks and is being released in the 4.3.18 z-stream release today. Since this is a direct cherrypick of #19831, I'm merging now.

**Pull request merged**